draft-ietf-teep-architecture.md

```diff
@@ -1347,6 +1347,33 @@ still a threat present in allowing a potentially compromised TAM to install
 any TA it wants; for example, a known-compromised TA or an explicitly
 malicious TA.
 
+The device manufacturer may want to include a notification support that can allow
+Device Administrators be notified for incidents upon detection with some mitigation 
+methods as suggested above.
```

Comment:
A compromised TAM can in theory compromise a TEE, and once you compromise a TEE, you may be able to prevent any such notifications from occurring by that device.  And outside the device, the device manufacturer may not be the right entity to provide things. So I don't understand how this sentence is actionable at present.

Reply:
I am thinking the common case where a TEE is embedded by a device manufacturer, and multiple TAMs, by some initial set of Trust Anchors. A compromised TAM can only manage TAs, not be able to compromise a TEE, in my view. It may attack a TEE environment but DoS - install many TAs etc but cannot corrupt TEE internals. A notification scheme could be, for example, some APIs that the TEE provides, allowing some others to build a notification app inside REE to get detection output in TEE. Both the detection and such monitoring APIs of any detection of incidents are out of scope of TEEP.

There may be other ways to get detected results when the detection is carried out inside the TEE, as we recommended use of method 1 with an ACL.

Reply:
> I am thinking the common case where a TEE is embedded by a device manufacturer, and multiple TAMs, by some initial set of Trust Anchors.

I am not sure that is a common case for TEEP.

> A compromised TAM can only manage TAs, not be able to compromise a TEE, in my view.

Disagree.  Maybe in a particular implementation, but I'm not sure that would be common.
A TAM has to be able to update dependencies, like OP-TEE, TF-A, etc.  Any Trusted Component.

> It may attack a TEE environment but DoS - install many TAs etc but cannot corrupt TEE internals.

"Cannot" is just an example, we should not make any claim as to whether in general one can or cannot, as we cannot say what will be "common" with TEEP yet, and it sounds like we have different expectations on what will be common.

> A notification scheme could be, for example, some APIs that the TEE provides, allowing some others to build a notification app inside REE to get detection output in TEE.

I don't know what "detection output in TEE" means.  Any detection in the REE cannot have visibility into things in the TEE or the TEE would fail the definition of a TEE in terms of code confidentiality or data confidentiality.

> Both the detection and such monitoring APIs of any detection of incidents are out of scope of TEEP.

Agree, but I'm also not convinced it's actually feasible or worth noting in this architecture document that already passed WGLC, IETF LC, and IESG ballot.

> There may be other ways to get detected results when the detection is carried out inside the TEE, as we recommended use of method 1 with an ACL.

I disagree we should be adding any recommendation at this stage as that would reset all the last call I think, and potentially take a while to get to consensus on.

Reply:
Removed as this is controversial, and it can be considered out of scope. Monitoring of TAM and TA activities is usually out of scope for a protocol design spec.